# Titanic ML Model to Predict Survival

Below I will walk through my process of building a machine learning model to predict the survival of passengers aboard the Titanic. Please let me know if you have any suggestions on how to improve my model, or any questions about my work. Thanks for stopping by!

# 1. Importing Data <a class="anchor"  id="chapter1"></a>


In [1]:
# Importing standard libraries
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 

# Importing and combining data, taking preliminary look 
train = pd.read_csv('/kaggle/input/titanic/train.csv')
test = pd.read_csv('/kaggle/input/titanic/test.csv')

# Combining data
combined = pd.concat([train, test], axis=0, ignore_index=True)

# Checking data types
display(combined.describe())

PassengerId    Survived       Pclass          Age        SibSp  \
count  1309.000000  891.000000  1309.000000  1046.000000  1309.000000   
mean    655.000000    0.383838     2.294882    29.881138     0.498854   
std     378.020061    0.486592     0.837836    14.413493     1.041658   
min       1.000000    0.000000     1.000000     0.170000     0.000000   
25%     328.000000    0.000000     2.000000    21.000000     0.000000   
50%     655.000000    0.000000     3.000000    28.000000     0.000000   
75%     982.000000    1.000000     3.000000    39.000000     1.000000   
max    1309.000000    1.000000     3.000000    80.000000     8.000000   

             Parch         Fare  
count  1309.000000  1308.000000  
mean      0.385027    33.295479  
std       0.865560    51.758668  
min       0.000000     0.000000  
25%       0.000000     7.895800  
50%       0.000000    14.454200  
75%       0.000000    31.275000  
max       9.000000   512.329200

The training set has 891 instances, the test set has 418 instances, and the combined has 1309. 